src/main/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImpl.java

```diff
@@ -197,4 +199,34 @@ public void setMaxChunkSizeToHsm(int maxChunkSizeToHsm) {
     public void setStopSending() {
         this.isStopped = true;
     }
+
+    @Override
+    public PowHSMBlockchainParameters getBlockchainParameters() throws HSMClientException {
+        final String BLOCKCHAIN_PARAMETERS_COMMAND = "blockchainParameters";
+        final String PARAMETERS_FIELD = "parameters";
+        final String CHECKPOINT_FIELD = "checkpoint";
+        final String MINIMUM_DIFFICULTY_FIELD = "minimum_difficulty";
+        final String NETWORK_FIELD = "network";
+
+        if (this.version < 3) {
+            throw new HSMUnsupportedTypeException("method call not allowed for version {}." + this.version);
+        }
+
+        ObjectNode command = this.hsmClientProtocol.buildCommand(BLOCKCHAIN_PARAMETERS_COMMAND, this.version);
+        JsonNode response = this.hsmClientProtocol.send(command);
+
+        this.hsmClientProtocol.validatePresenceOf(response, PARAMETERS_FIELD);
+        JsonNode parameters = response.get(PARAMETERS_FIELD);
+
+        this.hsmClientProtocol.validatePresenceOf(parameters, CHECKPOINT_FIELD);
+        this.hsmClientProtocol.validatePresenceOf(parameters, MINIMUM_DIFFICULTY_FIELD);
+        this.hsmClientProtocol.validatePresenceOf(parameters, NETWORK_FIELD);
+
+        String checkpoint = parameters.get(CHECKPOINT_FIELD).asText();
+        int minimumDifficulty = parameters.get(MINIMUM_DIFFICULTY_FIELD).asInt();
```

Comment:
Will this value fit in an `int`? What size can it have?

Reply:
See https://github.com/rsksmart/powpeg-node/blob/hsm3-refactors-integration/src/main/java/co/rsk/federate/config/PowHSMBookkeepingConfig.java#LL32C41-L32C51

Reply:
No It Wont, Refactored It To a BigInteger